### PR TITLE
Add recording file location

### PR DESCRIPTION
**Pull request by billwert**
There are tests in the Java repo which parse the recording files for various reasons. Add a header on the response of playback start to share that location.

I also noticed that the tests were using the wrong assertion for headers, so I changed it to check for string empty/null. (Even null is unnecessary - could be a test against `String.Empty` if you prefer.)

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs [hunk 1]
```diff
@@ -62,9 +62,9 @@ public async Task TestStartRecordInMemory()
 
         [Theory]
         [InlineData("recordings/TestStartRecordSimple_nosave.json", "request-response")]
-        [InlineData("recordings/TestStartRecordSimpl�_nosave.json", "request-response")]
+        [InlineData("recordings/TestStartRecordSimplé_nosave.json", "request-response")]
         [InlineData("recordings/TestStartRecordSimple.json", "")]
-        [InlineData("recordings/TestStartRecordSimpl�.json", "")]
+        [InlineData("recordings/TestStartRecordSimplé.json", "")]
```
Comment (billwert) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs:67]: I just noticed this - VS must have "helped." Should this get reverted?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs [hunk 2]
```diff
@@ -372,7 +372,7 @@ public async Task StartPlaybackAsync(string sessionId, HttpResponse outgoingResp
             {
                 await RestoreAssetsJson(assetsPath, true);
                 var path = await GetRecordingPath(sessionId, assetsPath);
-
+                outgoingResponse.Headers.Add("x-recording-file-location", path);
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs:375]: We should probably do the same in `StartRecordingAsync` for consistency right?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md [hunk 3]
```diff
@@ -351,6 +351,8 @@ To start an individual test recording or playback, users will POST to a route on
 
 You will receive a recordingId in the reponse under header `x-recording-id`. This value should be included under header `x-recording-id` in all further requests.
 
+For playback, you will receive the location of the recording file (if it is on disk) in the header `x-recording-file-location`.
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md:354]: If we're going to do this we might as well do it for both `Record/Start` and `Playback/Start` I think.

**Review by scbedd — APPROVED** [review]
Resolve the test and I'll approve. I see where you're coming from.

**Review by scbedd — APPROVED** [review]
Thanks @billwert
